packages/eslint-plugin/src/rules/prefer-ts-expect-error.ts

```diff
@@ -8,44 +15,58 @@ export default util.createRule<[], MessageIds>({
   meta: {
     type: 'problem',
     docs: {
-      description:
-        'Recommends using `// @ts-expect-error` over `// @ts-ignore`',
+      description: 'Recommends using `@ts-expect-error` over `@ts-ignore`',
       category: 'Best Practices',
       recommended: false,
     },
     fixable: 'code',
     messages: {
       preferExpectErrorComment:
-        'Use "// @ts-expect-error" to ensure an error is actually being suppressed.',
+        'Use "@ts-expect-error" to ensure an error is actually being suppressed.',
     },
     schema: [],
   },
   defaultOptions: [],
   create(context) {
-    const tsIgnoreRegExp = /^\/*\s*@ts-ignore/;
+    const tsIgnoreRegExp = /(^|\n)((\*)*(\/)*\s*(\/\*)*\s*)*\s*@ts-ignore/;
     const sourceCode = context.getSourceCode();
 
+    function getLastCommentLine(comment: TSESTree.Comment): string {
+      if (comment.type === AST_TOKEN_TYPES.Line) {
+        return comment.value;
+      }
+
+      // For multiline comments - we look at only the last line.
+      const commentlines = comment.value.split('/\n');
+      return commentlines[commentlines.length - 1];
+    }
     return {
       Program(): void {
         const comments = sourceCode.getAllComments();
-
         comments.forEach(comment => {
-          if (comment.type !== AST_TOKEN_TYPES.Line) {
-            return;
-          }
+          if (tsIgnoreRegExp.test(getLastCommentLine(comment))) {
+            const lineCommentRuleFixer = (fixer: RuleFixer): RuleFix =>
+              fixer.replaceText(
+                comment,
+                `//${comment.value.replace('@ts-ignore', '@ts-expect-error')}`,
+              );
+
+            const blockCommentRuleFixer = (fixer: RuleFixer): RuleFix =>
+              fixer.replaceText(
+                comment,
+                `/*${comment.value.replace(
```

Comment:
Will this work properly if you have something like `/*** @ts-ignore ***/` ? Is there a way to replace just the `@ts-ignore` part without changing the comment itself?

Reply:
it actually does - the comment source becomes - `** @ts-ignore **`

Reply:
it does - the comment we get because of `sourceCode.getAllComments` is stripped of the initial `/*`

So in this example we get `** @ts-ignore **` instead of and as expected `/*** @ts-ignore ***/`

Reply:
but we might end up replacing the range - will wait for consensus on other discussions to proceed ahead.

Reply:
I have aligned it with the tsRegex with some changes to accommodate how we get comments